src/effects/liftEffects.ts

```diff
@@ -0,0 +1,63 @@
+import { Effect } from "./Effect"
+import { Observable, Subscriber } from "rxjs"
+import type { EffectObservable } from "../index.d"
+
+type IsEmpty<T> = unknown extends T ? true : T extends never ? true : false
+
+export function liftEffects(): <T, E>(
+  source$: EffectObservable<T, E>,
+) => EffectObservable<IsEmpty<E> extends true ? T : T | E, never>
+export function liftEffects<Args extends Array<unknown>>(
```

Comment:
These types are duplicate - on `index.d.ts` and here... should we somehow merge them?